src/Diagnostics/hyperdiffusion_balancelaw.jl

```diff
@@ -0,0 +1,171 @@
+# A mini balance law for computing hyperdiffusion using the DG kernels.
+#
+
+using StaticArrays
+using UnPack
+
+using ..BalanceLaws
+using ..MPIStateArrays
+using ..Orientations
+using ..VariableTemplates
+
+import ..BalanceLaws:
+    vars_state,
+    flux_first_order!,
+    flux_second_order!,
+    source!,
+    eq_tends,
+    flux,
+    source,
+    wavespeed,
+    boundary_conditions,
+    boundary_state!,
+    compute_gradient_argument!,
+    compute_gradient_flux!,
+    transform_post_gradient_laplacian!,
+    init_state_auxiliary!,
+    init_state_prognostic!,
+    update_auxiliary_state!,
+    indefinite_stack_integral!,
+    reverse_indefinite_stack_integral!,
+    integral_load_auxiliary_state!,
+    integral_set_auxiliary_state!,
+    reverse_integral_load_auxiliary_state!,
+    reverse_integral_set_auxiliary_state!
+
+
+# A mini balance law that is used to take the gradient of u and v
+# to obtain hyperdiffusion.
+struct DryBiharmonicModel{PS, O, FT} <: BalanceLaw
+    param_set::PS
+    orientation::O
+    τ_timescale::FT
+end
+
+vars_state(::DryBiharmonicModel, ::Prognostic, FT) =
+    @vars(hyper_e::SVector{3, FT}, hyper_u::SMatrix{3, 3, FT, 9})
+function vars_state(h::DryBiharmonicModel, st::Auxiliary, FT)
+    @vars begin
+        Δ::FT
+        ρ::FT
+        ρu::SVector{3, FT}
+        ρe::FT
+        orientation::vars_state(h.orientation, st, FT)
+    end
+end
+vars_state(::DryBiharmonicModel, ::Gradient, FT) =
+    @vars(u_h::SVector{3, FT}, h_tot::FT)
+vars_state(::DryBiharmonicModel, ::GradientLaplacian, FT) =
+    @vars(u_h::SVector{3, FT}, h_tot::FT)
+vars_state(::DryBiharmonicModel, ::Hyperdiffusive, FT) =
+    @vars(ν∇³u_h::SMatrix{3, 3, FT, 9}, ν∇³h_tot::SVector{3, FT})
+
+# required for each balance law 
+function init_state_auxiliary!(
+    h::DryBiharmonicModel,
+    state_auxiliary::MPIStateArray,
+    grid,
+    direction,
+)
+    init_aux!(h, h.orientation, state_auxiliary, grid, direction)
+    init_state_auxiliary!(
+        h,
+        nodal_init_state_auxiliary!,
+        state_auxiliary,
+        grid,
+        direction,
+    )
+end
+function init_state_prognostic!(
+    h::DryBiharmonicModel,
+    state::Vars,
+    aux::Vars,
+    localgeo,
+    t,
+) end
+#=
+#function nodal_update_auxiliary_state!(
+    h::DryBiharmonicModel,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+) end =#
+
+function nodal_init_state_auxiliary!(
+    h::DryBiharmonicModel,
+    aux::Vars,
+    tmp::Vars,
+    geom::LocalGeometry,
+)
+    aux.Δ = lengthscale(geom)
+end
+
+function compute_gradient_argument!(
+    h::DryBiharmonicModel,
+    transform::Vars,
+    state::Vars,
+    aux::Vars,
+    t::Real,
+)
+    ρinv = 1 / aux.ρ
+    u = aux.ρu * ρinv
+    k̂ = vertical_unit_vector(h.orientation, h.param_set, aux)
+    u_h = (SDiagonal(1, 1, 1) - k̂ * k̂') * u
+    transform.u_h = u_h
+    transform.h_tot = transform.h_tot
```

Comment:
I just realized  [here](https://github.com/CliMA/ClimateMachine.jl/blob/17152cfa3f2f7c909931882d2ecdad2135bb41c6/src/Common/TurbulenceClosures/TurbulenceClosures.jl#L925-L926) in the `TurbulenceClosure.jl`. Not sure if this is related..

Reply:
Fixed this.